### PR TITLE
Add -w or --workdir to compose run to override workdir from commandline

### LINKED ISSUE #2811: add -w--workdir=  option to docker-compose run
Same as docker

### PR DESCRIPTION
**Pull request by simonvanderveldt**
This adds the possibility to override the workdir from the command line, using either `-w` or `--workdir` just like `docker run`.
In essence it does the same as #332, though that code couldn't really be rebased so I wrote my own. Do we want to attribute anything of this to @scottynomad?

Also not really sure about the acceptance tests, there seem to be multiple different ways of doing them within `tests/acceptance/cli_test.py`. I basically mirrored the `-u` / `--user` tests.

Relevant for #363 
Fixes #2811

### CODE REVIEW

**Comment by simonvanderveldt** [comment]
@GordonTheTurtle doesn't seem to be happy because the DinD tag used in the test is missing
```
++ docker run -d --name compose-dind-1.10.2-rc1-2391 --privileged --volume=/var/lib/docker dockerswarm/dind:1.10.2-rc1 docker daemon -H tcp://0.0.0.0:2375 --storage-driver=aufs
Unable to find image 'dockerswarm/dind:1.10.2-rc1' locally
Pulling repository docker.io/dockerswarm/dind
Tag 1.10.2-rc1 not found in repository docker.io/dockerswarm/dind
```

**Comment by dnephin** [comment]
We have a PR #2594 which adds this already, and I think it's ready to merge

I'm fixing the dind image build now

**Comment by simonvanderveldt** [comment]
I noticed there were markdown files documenting the commands as well, so I've updated `docs/reference/run.md` to reflect the change in this PR.

The test on Janky now fails on the flake8 test, which works fine locally. Not sure why this is happening.
Note the test on Janky passed before I added the markdown file, so I guess something is broken/has changed in the test?

**Comment by dnephin** [comment]
I've got a fix for the max-complexity error in this commit: https://github.com/dnephin/compose/commit/8246c2afd5b09e612e0a5a6806baa6b909299d7e

I guess I rebased on master, so it doesn't merge cleanly on your branch, but you can cherry-pick this commit if you add my fork as a remote.

**Comment by simonvanderveldt** [comment]
> I've got a fix for the max-complexity error in this commit: dnephin@8246c2a

Thanks! 

> I guess I rebased on master, so it doesn't merge cleanly on your branch, but you can cherry-pick this commit if you add my fork as a remote.

Done, works fine locally, hope Janky is happy this time :)

File: compose/cli/main.py [hunk 1]
```diff
@@ -472,6 +472,7 @@ def run(self, project, options):
                                   to the host.
             -T                    Disable pseudo-tty allocation. By default `docker-compose run`
                                   allocates a TTY.
+            -w, --workdir=""      Override the workdir
```
Comment (dnephin) [inline compose/cli/main.py]: I just noticed, this should match what `docker run --help` says "Working directory inside the container"

**Comment by dnephin** [comment]
With the fix to the help text LGTM

File: compose/cli/main.py (hunk 1 above)
Comment (simonvanderveldt) [inline compose/cli/main.py]: > I just noticed, this should match what docker run --help says "Working directory inside the container"

Done, thx for the hint!

**Comment by dnephin** [comment]
LGTM

**Comment by simonvanderveldt** [comment]
Janky errors seem to be unrelated
```
____________________________ CLITestCase.test_kill _____________________________

tests/acceptance/cli_test.py:1163: in test_kill

    self.dispatch(['up', '-d'], None)

tests/acceptance/cli_test.py:128: in dispatch

    return wait_on_process(proc, returncode=returncode)

tests/acceptance/cli_test.py:48: in wait_on_process

    assert proc.returncode == returncode

E   AssertionError: assert 1 == 0

E    +  where 1 = <subprocess.Popen object at 0x7f30c77aacd0>.returncode

----------------------------- Captured stdout call -----------------------------

Running process: 205

Stderr: Creating simplecomposefile_simple_1

Starting simplecomposefile_another_1

Container is marked for removal and cannot be started.



Stdout: 

----------------------------- Captured stderr call -----------------------------

Killing simplecomposefile_simple_1 ... 



Killing simplecomposefile_simple_1 ... done
Removing simplecomposefile_simple_1 ... 


Removing simplecomposefile_another_1 ... 



Removing simplecomposefile_simple_1 ... done

Removing simplecomposefile_another_1 ... error


ERROR: for simplecomposefile_another_1  Driver aufs failed to remove root filesystem f75f290d301456234f1d37c18bd63254a3f6bd48c3efcc2b099768760c071b3c: rename /var/lib/docker/aufs/mnt/0cb124ee817b6072798a141ba784ce4c79f427269198d412424bd8117fdd58e8 /var/lib/docker/aufs/mnt/0cb124ee817b6072798a141ba784ce4c79f427269198d412424bd8117fdd58e8-removing: device or resource busy 
```

**Comment by dnephin** [comment]
yup, unrelated

**Comment by dnephin** [comment]
Needs a rebase to be merged

**Comment by simonvanderveldt** [comment]
> Needs a rebase to be merged

@dnephin When will you need it to be rebased? I'm on holidays till the 25th of march :)

**Comment by dnephin** [comment]
Ok, don't worry about it. I will pick this up and carry the commits for the release. Code freeze is this week.

**Comment by dnephin** [comment]
Carried in #3136

**Comment by simonvanderveldt** [comment]
Awesome, thanks!

**Pull request closed without merging**